### ISSUE TITLE
Stack overflow?

### ISSUE THREAD

**Issue by tdsfugal** [mostjs-community/most-proxy#5]
I've tried several ways to close the cycle with Proxy and they all seem to result in a stack overflow.  Perhaps this sets up an infinite loop in the stream control logic? 

This example counts to 5 the hard way, but it's a good example of the process I'm trying to model (shutting down a harmonic oscillator when it comes to rest).  y$ emits a single true when the accumulated sum reaches 15.  Closed-loop this would shut x$ down.

When the attach line is commented out you can see this in the console log.  When it not commented out Chrome just sits there for a few seconds until devtools disconnects. I presume this crash is from the stack filling up with an endless stream of handshaking. 

```
import { proxy }     from 'most-proxy'
import { periodic }  from 'most'

const trigger$ = periodic(1000, 1)
  .scan( (s, t) => s + t, 0)
  .take(10)

const { stream: yProxy$, attach: attachY } = proxy()

const x$ = trigger$.until(yProxy$)

const y$ = x$
  .scan( (s, x) => s + x, 0 )
  .map( sum => sum >= 15)
  .filter( b => b )
  .skipRepeats()

attachY(y$)

x$.observe( x => console.log(x))
y$.observe( x => console.log(x))
```

[edit] - Perhaps some sort of air-gap or break in the handshaking protocols is needed in the proxy stream to make the stream graph appear acyclic to the scheduler?

**Comment by TylorS** [comment]
This is probably related to #4

**Comment by TylorS** [comment]
Hi @tdsfugal sorry it's been a while, but there is a fix for #4 now, and that was indeed the issue here.

http://www.webpackbin.com/Nk5JUo1WG

Feel free to reopen if that's not the correct output.
